File: Clase semana 05/02 - Clases grabadas/2024/1C/Clase05-Lunes/src/Complejo.cpp
/*
 * Complejo.cpp
 *
 *  Created on: 08/04/2024
 *      Author: algo2
 */

#include "Complejo.h"
#include <iostream>

Complejo::Complejo() {
	this->parteReal = 0.0;
	this->parteImaginaria = 0.0;
	cantidadDeComplejos++;
}

Complejo::Complejo(Complejo &complejo) {
	this->parteReal = complejo.parteReal;
	this->parteImaginaria = complejo.parteImaginaria;
}

Complejo * Complejo::sumar(Complejo *&termino1, Complejo *&termino2) {
	Complejo * total = new Complejo(termino1->getParteReal(), termino1->getParteImaginaria());
	return total->sumar(termino2);
}

Complejo::Complejo(double parteReal, double parteImaginaria) {
	this->setParteReal(parteReal);
	this->parteImaginaria = parteImaginaria;
	cantidadDeComplejos++;
}

Complejo::~Complejo() {}

double Complejo::getParteImaginaria() {
	return this->parteImaginaria;
}

double Complejo::getParteReal() {
	return this->parteReal;
}

void Complejo::setParteImaginaria(double parteImaginaria) {
	this->parteImaginaria = parteImaginaria;
}

void Complejo::setParteReal(double parteReal) {
	this->parteReal = parteReal;
}

Complejo * Complejo::sumar(Complejo * complejo) {
	validar(complejo);
	this->parteReal += complejo->parteReal;
	this->parteImaginaria += complejo->parteImaginaria;
	return this;
}

void Complejo::validar(Complejo * &complejo) {
	if (complejo == NULL) {
		throw "EL parametro no puede ser nulo";
	}
}

File: Clase semana 03/02 - Clases grabadas/2024/2C/src/ar/uba/fi/cb100/semana03/miercoles/Complejo.java
package ar.uba.fi.cb100.semana03.miercoles;

public class Complejo {

	private double parteReal;
	private double parteImaginaria;

	public Complejo() {
		this.parteReal = 0;
		this.parteImaginaria = 0;
	}
	
	public Complejo(double parteReal) {
		this.parteReal = parteReal;
		this.parteImaginaria = 0;
	}
	
	public Complejo(String parteReal) {
		this.parteReal = Double.valueOf(parteReal);
		this.parteImaginaria = 0;
	}
	
	public Complejo(double parteReal, double parteImaginaria) {		
		this.parteReal = parteReal;
		this.parteImaginaria = parteImaginaria;
	}
	
	public double getParteReal() {
		return parteReal;
	}
	
	public void setParteReal(double parteReal) {
		this.parteReal = parteReal;
	}
	
	public double getParteImaginaria() {
		return parteImaginaria;
	}
	
	public void setParteImaginaria(double parteImaginaria) {
		this.parteImaginaria = parteImaginaria;
	}
	
	/**
	 * pre: complejo no puede ser nulo
	 * post: va a sumar en un nuevo complejo, el complejo actual y complejo por parametro.
	 * @param complejo: valor a sumar
	 * @return
	 * @throws Exception 
	 */
	public Complejo sumar(Complejo complejo) throws Exception {
		if (complejo == null) {
			throw new Exception("El parametro no puede ser nulo");
		}
		return new Complejo(this.parteReal + complejo.getParteReal(), this.parteImaginaria + complejo.getParteImaginaria());
	}
	
	/**
	 * pre: no hay  
	 * @param parteReal valor a sumar
	 * @return devuelve la suma del complejo actual mas la parte real. La parte imaginaria queda como esta. 
	 */
	public Complejo sumar(double parteReal) {
		return new Complejo(this.parteReal + parteReal, this.parteImaginaria);
	}

}
